src/components/oculus-touch-controls.js

```diff
@@ -0,0 +1,259 @@
+var registerComponent = require('../core/component').registerComponent;
+var bind = require('../utils/bind');
+var trackedControlsUtils = require('../utils/tracked-controls');
+
+const TOUCH_CONTROLLER_MODEL_BASE_URL = 'https://cdn.aframe.io/controllers/oculus/';
+const TOUCH_CONTROLLER_MODEL_OBJ_URL_L = TOUCH_CONTROLLER_MODEL_BASE_URL + 'oculus_cv1_controller_left.obj';
+const TOUCH_CONTROLLER_MODEL_OBJ_MTL_L = TOUCH_CONTROLLER_MODEL_BASE_URL + 'oculus_cv1_controller_left.mtl';
+const TOUCH_CONTROLLER_MODEL_OBJ_URL_R = TOUCH_CONTROLLER_MODEL_BASE_URL + 'oculus_cv1_controller_right.obj';
+const TOUCH_CONTROLLER_MODEL_OBJ_MTL_R = TOUCH_CONTROLLER_MODEL_BASE_URL + 'oculus_cv1_controller_right.mtl';
+
+const GAMEPAD_ID_PREFIX = 'Oculus Touch';
+
+const FAKE_TOUCH_THRESHOLD = 0.00001;
+
+/**
+ * Oculus Touch Controls Component
+ * Interfaces with Oculus Touch controllers and maps Gamepad events to
+ * common controller buttons: trackpad, trigger, grip, menu and system
+ * It loads a controller model and highlights the pressed buttons
+ */
+module.exports.Component = registerComponent('oculus-touch-controls', {
+  schema: {
+    hand: {default: 'left'},
+    buttonColor: { default: '#FAFAFA' },  // Off-white.
+    buttonHighlightColor: {default: '#22D1EE'},  // Light blue.
+    model: {default: true},
+    rotationOffset: {default: 0} // no default offset; -999 is sentinel value to auto-determine based on hand
+  },
+
+  // buttonId
+  // 0 - thumbstick
+  // 1 - trigger ( intensity value from 0.5 to 1 )
+  // 2 - grip
+  // 3 - menu ( dispatch but better for menu options )
+  // 4 - system ( never dispatched on this layer )
+  mapping: {
+    'left': {
+      axis0: 'thumbstick',
+      axis1: 'thumbstick',
+      button0: 'thumbstick',
+      button1: 'trigger',
+      button2: 'grip',
+      button3: ['oculus-touch.A-or-X', 'oculus-touch.X'],
+      button4: ['oculus-touch.Y', 'menu'],
+      button5: 'surface'
+    },
+    'right': {
+      axis0: 'thumbstick',
+      axis1: 'thumbstick',
+      button0: 'thumbstick',
+      button1: 'trigger',
+      button2: 'grip',
+      button3: ['oculus-touch.A-or-X', 'oculus-touch.A'],
+      button4: ['oculus-touch.B', 'menu'],
+      button5: 'surface'
+    }
+  },
+
+  init: function () {
+    var self = this;
+    this.animationActive = 'pointing';
+    this.onButtonChanged = bind(this.onButtonChanged, this);
+    this.onButtonDown = function (evt) { self.onButtonEvent(evt.detail.id, 'down'); };
+    this.onButtonUp = function (evt) { self.onButtonEvent(evt.detail.id, 'up'); };
+    this.onButtonTouchStart = function (evt) { self.onButtonEvent(evt.detail.id, 'touchstart'); };
+    this.onButtonTouchEnd = function (evt) { self.onButtonEvent(evt.detail.id, 'touchend'); };
+    this.onModelLoaded = bind(this.onModelLoaded, this);
```

Comment:
You can put all the calls to `bind(...)` in a method call `bindMethods`. This will clean up the `init` function